Package.swift

```diff
@@ -129,19 +129,46 @@ func isDebuggingMain(_ isDebug: Bool) {
     }
 }
 
+func xcodeFolder() -> String {
+    (Executable("/usr/bin/xcode-select")("-p") ?? "").trimmed
+}
+
+func swiftCompilerVersion() -> String {
+    let versionLine = Executable(
+        "\(xcodeFolder().trimmed)/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-frontend"
+    )("--version") ?? ""
+
+    let regex = try? NSRegularExpression(pattern: " \\d.\\d")
+    return regex?.firstMatch(in: versionLine, range: NSRange(location: 0, length: versionLine.count)).flatMap {
+        Range($0.range, in: versionLine)
```

Comment:
Did you check to see if there's a way to get Xcode-select to print out a well-structured/json version of this info?

Reply:
yeah, it's `swift-frontend` but I could not find anything of the sort :(